js/components/portal-dashboard/all-responses-popup/student-responses-popup.tsx

```diff
@@ -55,7 +72,25 @@ export class StudentResponsePopup extends React.PureComponent<IProps> {
           isAnonymous={isAnonymous}
           currentQuestion={currentQuestion}
         />
+        { showSpotlightDialog && (selectedStudents.length > 0
+          ? <SpotlightStudentListDialog
+              students={students}
+              anonymous={anonymous}
+              isAnonymous={isAnonymous}
+              currentActivity={currentActivity}
+              currentQuestion={currentQuestion}
+              setAnonymous={setAnonymous}
+              handleCloseSpotlightStudentListDialog={this.setShowSpotlightDialog}
+            />
+          : <NoStudentSelectedSpotlightDialog
+              handleCloseNoStudentSelectedDialog={this.setShowSpotlightDialog}
+            />)
+        }
       </div>
     );
   }
+
+  private setShowSpotlightDialog = (show: boolean) => {
+    this.setState({ showSpotlightDialog: show });
+  }
```

Comment:
We can write this with function currying as follows and then get away with some shorter code in other places:
```
  private setShowSpotlightDialog = (show: boolean) => () => {
    this.setState({ showSpotlightDialog: show });
  }
```

Reply:
Maybe rename this function simply: `showSpotlightDialog`